tls/s2n_connection.c

```diff
@@ -178,6 +181,7 @@ struct s2n_connection *s2n_connection_new(s2n_mode mode)
     conn->security_policy_override = NULL;
     conn->ticket_lifetime_hint = 0;
     conn->session_ticket_status = S2N_NO_TICKET;
+    conn->preferred_key_shares = 0;
```

Comment:
You don't need to set it to 0 here either. _new calls _wipe.

Reply:
This is the `init` call, and here we want to make sure that the value of `preferred_key_shares=0` and has no garbage value in it.

Reply:
Hmmm...doesn't the same argument hold true for these values as well https://github.com/awslabs/s2n/blob/master/tls/s2n_connection.c#L164-L179?